File: Ficheros texto_binario/ProductoBinario.py
import pickle

class ProductoBinario:
    _version = 1  

    def __init__(self, nombre: str, precio: float):
        self._nombre = nombre
        self._precio = precio

    def get_nombre(self):
        return self._nombre

    def set_nombre(self, nombre: str):
        self._nombre = nombre

    def get_precio(self):
        return self._precio

    def set_precio(self, precio: float):
        self._precio = precio

    def __str__(self):
        return f"ProductoBinario {{nombre = {self._nombre}, Precio = {self._precio} }}"

    @staticmethod
    def check_version(obj):
        if hasattr(obj, '_version') and obj._version == ProductoBinario._version:
            return True
        else:
            raise ValueError("La versión del objeto serializado no coincide con la versión actual de la clase")

File: Hilos Java/Ficheros texto_binario/Producto_Binario.java
import java.io.Serializable;

public class Producto_Binario implements Serializable {
    private static final long serialVersionUID = 1L;
    private String nombre;
    private float precio;

    public Producto_Binario(String nombre, float precio) {
        this.nombre = nombre;
        this.precio = precio;

    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public float getPrecio() {
        return precio;
    }

    public void setPrecio(float precio) {
        this.precio = precio;
    }


    @Override
    public String toString(){
        return "Producto {nombre ="+ nombre +", Precio="+ precio+" }";
        }
}